orchagent/neighorch.cpp

```diff
@@ -331,6 +361,15 @@ bool NeighOrch::removeNeighbor(NeighborEntry neighborEntry)
     SWSS_LOG_NOTICE("Removed next hop %s on %s",
                     ip_address.to_string().c_str(), alias.c_str());
 
+    if (neighbor_entry.ip_address.addr_family == SAI_IP_ADDR_FAMILY_IPV4)
```

Comment:
This is tricky. When status is SAI_STATUS_ITEM_NOT_FOUND, we don't have to update resource counter for NEXTHOP. Guess its better to check for status here. Also note below, for the NEIGBHOR case, if ITEM_NOT_FOUND, there is no update to resource counter